File: sas_token_generation/ObjectiveC/SasTokensObjectiveC/SasTokensObjectiveC/TokenData.m
//
//  TokenData.m
//  SasTokensObjectiveC
//
//  Created by Mike Parker on 26/11/2018.
//  Copyright © 2018 mobcat. All rights reserved.
//

#import <Foundation/Foundation.h>

typedef struct {
    NSString* token;
    NSDate* expiration;
} TokenData;

TokenData InitTokenDataWithTokenAndExpiration(NSString* token, NSDate* expiration) {
    
    TokenData tokenData;
    tokenData.token = token;
    tokenData.expiration = expiration;
    
    return tokenData;
}

File: sas_token_generation/Swift/SasTokensSwift/SasTokensSwift/TokenData.swift
//
//  TokenData.swift
//  SasTokensSwift
//
//  Created by Mike Parker on 26/11/2018.
//  Copyright © 2018 mobcat. All rights reserved.
//

import Foundation

struct TokenData {
    
    let token : String
    let expiration : Date
    
    init(withToken token : String, andTokenExpiration expiration : Date) {
        self.token = token
        self.expiration = expiration
    }
}
